### DISCUSSION TITLE
Call to an undefined method Laravel\Socialite\Contracts\Provider::stateless().

### DISCUSSION THREAD

**Discussion by tcampbPPU** [larastan/larastan#1601] [Q&A]
I am using Laravel Socialite following the [docs](https://laravel.com/docs/10.x/socialite#stateless-authentication) when using stateless the code runs/works fine but when I run larastan, I get the following error:
```
Call to an undefined method Laravel\Socialite\Contracts\Provider::stateless()
```

**Comment by tcampbPPU** [comment]
Right now I am just ignoring in my config but would like to fix!
```neon
includes:
    - ./vendor/nunomaduro/larastan/extension.neon

parameters:
    paths:
        - app
    level: 8
    ignoreErrors:
      - '#Call to an undefined method Laravel\\Socialite\\Contracts\\Provider::stateless\(\)\.#'
    excludePaths:
    checkMissingIterableValueType: false
```

**Comment by szepeviktor** [comment]
`stateless` is available on the implementation, not on the contract (interface)

**Reply by tcampbPPU** [reply]
does this mean the import is wrong, I am using
```
use Laravel\Socialite\Facades\Socialite;
```
like the docs show?

**Reply by szepeviktor** [reply]
Larastan somehow does not see through the facade.
@canvural Could you help?

**Reply by mad-briller** [reply]
the package is missing a `@mixin` annotation on the facade:
https://github.com/laravel/socialite/blob/5.x/src/Facades/Socialite.php#L8-L12

i imagine they'd accept a pr which added it

**Reply by szepeviktor** [reply]
https://github.com/nunomaduro/larastan/blob/11bdd132565d37e808e4ac985a580b492c2d8887/src/Methods/Pipes/Facades.php#L41

Larastan currently sees through facades but not this one.

**Reply by mad-briller** [reply]
ah interesting, i thought it was relying on the comments above modern laravel facades:
https://github.com/laravel/framework/blob/10.x/src/Illuminate/Support/Facades/App.php

looks like they changed from using `@mixin` to just defining everything upfront aswell

**Reply by mad-briller** [reply]
hmm actually @szepeviktor i think your initial point is correct, `->stateless()` doesn't exist on the `Provider` interface, and `->driver()` returns a `Provider` as per the comment above the facade

larastan doesn't know that the driver that is being returned supports `stateless()`, which would be hard to tell statically tbh

**Comment by canvural** [comment]
Currently we do not have plans to support any Laravel packages. Also like others said, the error makes sense. `stateless` does not exist on the contract.
